include/aie/Dialect/AIE/IR/AIEOps.td

```diff
@@ -835,13 +863,21 @@ def AIE_DMABDOp: AIE_Op<"dma_bd", []> {
   let hasVerifier = 1;
 
   let assemblyFormat = [{
-    `(` $buffer `:` type($buffer) (`,` $offset^)? (`,` $len^)? (`,` $dimensions^)? `)` attr-dict
+    `(` $buffer `:` type($buffer) (`,` `dims` `=` $dims^)? `)` attr-dict
```

Comment:
Do we still need `dims` in assemblyFormat? You also add "dims" as an attribute on line 894 below, does that make it part of `attr-dict`?

Reply:
All attributes are always in the attribute dictionary (on the C++ side) but the reason I put dims here is because if you put them into `attr-dict` (i.e., tell the parser to parse them from there) then you have to annotate each size and stride with their type (`size: i32`, etc) where here you don't need to. You can do some extra work to actually get `i32`s automatically "inferred" even in the `attr-dict` (and I tried) but I just don't want to do all of that.